development/process.md

```diff
@@ -101,6 +102,94 @@ keep the amount of rework from the contributor to a minimum.
+functional changes like adding a new feature or modifying code behaviour. This
+makes reviewing the code much easier and reduces the chance of introducing
+unintended changes in behavior.
+
+Whenever in doubt on splitting a change into a separate commit, ask yourself the
+following question: if all other work in the PR needs to be reverted after
+merging to master for some objective reason, such as a bug has been discovered,
+is it worth keeping that commit still in master.
+
+When writing a commit message, follow the seven rules of a great commit message:
+
+* Separate subject from body with a blank line.
+* Limit the subject line to 50 characters.
+* Capitalize the subject line
+* Do not end the subject line with a period.
+* Use the imperative mood in the subject line.
+* Wrap the body at 72 characters.
+* Use the body to explain what and why versus how.
+
+Read the [full commit message guide](https://chris.beams.io/posts/git-commit/)
+for more details and examples.
+
+## Release note guidelines <a name="release-note"></a>
+
+Release notes should communicate the user-facing aspects of all change with a
+specific version. As a results internal refactoring or improvements, such as
+build fixes or new tests, are not included. New features, new configuration,
+performance changes, bug fixes and other aspects relevant for users and
+administrators must be included.
+
+Release note entries should be suggested by the contributor. Correct content and
+wording is part of the pull request review and merge process. If necessary the
+contributor and merging maintainer can work with the maintainer assembling the
+release notes PR.
+
+Some release notes are broken up into separate section for different components.
+For example, Trino release are ordered following the
+[template](https://github.com/trinodb/trino/blob/master/docs/release-template.md)
+
+In each section different release notes entries are sorted:
+
+1. **New features**: Start with `Add` or `Add support for` or similar wording
+2. **Performance improvements**: Start with `Improve` or `Improve performance`
+   or similar wording
+3. **Bug fixes**: Start with `Fix` or `Prevent`  or similar wording
+
+Use imperative present to describe change. When a change adds configuration, add
+the configuration details after the description of the functionality. Detail the
+property names and their types. Consider linking to new documentation.
+
+View older release notes for example and strive for consistency with other
+release notes and release notes entries.
+
+Link to specific sections in the documentation to avoid overly long entries,
+especially if more information is needed.
+
+Release notes entries that prevent application start up with old configuration
+or otherwise change application behavior significantly must be marked as
+breaking changes. Details vary based on the documentation system, such as Sphinx
+in Trino and mkdocs in Trino Gateway.
```

Comment:
The "Details vary" sentence needs help.  Also we should mention above that release notes are written in Markdown

Reply:
Expanded and fixed.